pkg/daemon/certs/certs.go

```diff
@@ -0,0 +1,125 @@
+package certs
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"syscall"
+	"unsafe"
+
+	syswindows "golang.org/x/sys/windows"
+	"k8s.io/klog/v2"
+)
+
+// Reconcile ensures the certificates in the given trusted CA bundle file are imported as system certificates.
+// Returns a boolean whether any certs have needed to be imported
+func Reconcile(trustBundle string) (bool, error) {
```

Comment:
Can this function be made readable? Can you have a trust bundle object that you act upon?

Reply:
need to explore this more

Reply:
Introduced more helper functions to keep this top level func more readable